lib/puppet/module.rb

```diff
@@ -52,6 +53,30 @@ def self.is_module_namespaced_name?(name)
     return false
   end
 
+  # @api private
+  def self.parse_range(range, strict)
+    @parse_range_method ||= SemanticPuppet::VersionRange.method(:parse)
+    if @parse_range_method.arity == 1
+      @semver_gem_version ||= SemanticPuppet::Version.parse(SemanticPuppet::VERSION)
+
+      # Give user a heads-up if the desired strict setting cannot be honored
+      if strict
+        if @semver_gem_version.major < 1
+          Puppet.warn_once('strict_version_ranges', 'version_range_cannot_be_strict',
+            _('VersionRanges will never be strict when using non-vendored SemanticPuppet gem, version %{version}') % { version: @semver_gem_version} )
+        end
+      else
+        if @semver_gem_version.major >= 1
+          Puppet.warn_once('strict_version_ranges', 'version_range_always_strict',
+            _('VersionRanges will always be strict when using non-vendored SemanticPuppet gem, version %{version}') % { version: @semver_gem_version} )
```

Comment:
To "resolve" this warning a user would be forced to run with `--strict_semver`, even though they may not be anything wrong with the versions. That seems overly invasive for little gain.